### PR TITLE
feat: typescript typings and example

### PR DESCRIPTION
**Pull request by JeremiGendron**
fixes #486 Also fix up the type declarations which were provided by @NickBolles
And import '@nuxt/auth' relatively in the typescript example's package.json

### CODE REVIEW

File: examples/demo/pages/login.vue [hunk 1]
```diff
@@ -83,31 +83,34 @@ export default {
...
+        await this.$auth
+          .loginWith('local', {
+            data: {
+              username: this.username,
+              password: this.password
+            }
+          })
+
+      } catch(e) {
+        this.error = e + ''
```
Comment (JoaoPedroAS51) [inline examples/demo/pages/login.vue:96]: Just see that this line has't been merged. Can you change to `this.error = e.response.data + ''` please :)

File: examples/demo/pages/login.vue [hunk 2]
```diff
@@ -83,31 +83,34 @@ export default {
...
           data: {
             username: this.username,
             password: this.password
           }
         })
-        .catch(e => {
-          this.error = e.response.data + ''
-        })
+      } catch(e) {
+          this.error = e + ''
```
Comment (JoaoPedroAS51) [inline examples/demo/pages/login.vue:112]: Same here.

File: examples/with-typescript/mixins/auth.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+   * @returns {Promise<void>}
+   */
+  async logout () {
+
+    this.$toast.show('Logging out...')
+    try {
+      await this.$auth.logout()
+      this.$toast.success('Successfully disconnected')
+    } catch(err) {
+      this.$toast.error('Error while disconnecting: ' + err.message)
```
Comment (JoaoPedroAS51) [inline examples/with-typescript/mixins/auth.ts:32]: And can you change `err.message` to `err.response.data`?

File: examples/with-typescript/mixins/auth.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    try {
+      await this.$auth.login({
+        data: {
+          username: this.username,
+          password: this.password
+        }
+      })
+      this.$toast.success('Successfully connected')
+    } catch(err) {
+      this.$toast.error('Error while connecting: ' + err.message)
```
Comment (JoaoPedroAS51) [inline examples/with-typescript/mixins/auth.ts:55]: Same here.

File: examples/with-typescript/nuxt.config.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  dev: false,
+  render: {
+    resourceHints: false
+  },
+  serverMiddleware: ["../api/auth"],
+  auth: {
+    strategies: {
+      local: {
+        endpoints: {
+          login: { propertyName: "token.accessToken" }
```
Comment (JoaoPedroAS51) [inline examples/with-typescript/nuxt.config.ts:19]: This is deprecated. Now we set token property like this:

```js
auth: {
    strategies: {
      local: {
        token: {
          property: "token.accessToken"
        }
      },
    }
  },
```

**Comment by NickBolles** [comment]
@JeremiGendron @JoaoPedroAS51 nice work. Looks good! sorry for the delay on #486 I'll go ahead and close that now.

**Comment by NickBolles** [comment]
can this be closed now since #621  merged?

**Comment by pi0** [comment]
Thanks for PR @JeremiGendron :heart:

**Pull request closed without merging**